gray_model.py

```diff
@@ -208,5 +209,8 @@ def forward(self, x):
         out= self.subpixel(out)
         out = self.conv_output(out)
         out = torch.add(out, residual)
```

Comment:
Watch out here there is already a residual connection: make sure this is only executed if we are not in exact recon mode.